In [1]:
from sympy import *

z = 2 ## digits in float numbers

In [2]:
a, b, c, d, e, f, g, h, i, j,I2, I4, I5, I6, I7, I8, I9, I10, I11, I12, I13, I15, I16, I17, I20, I21, I22, I23, I24, I25, I26, I27, I28, I29, I30, I31, I32, I37, I38, I39, I40, I41, I42, I43, I44, I45, I46, I48 = symbols("a b c d e f g h i j I2 I4 I5 I6 I7 I8 I9 I10 I11 I12 I13 I15 I16 I17 I20 I21 I22 I23 I24 I25 I26 I27 I28 I29 I30 I31 I32 I37 I38 I39 I40 I41 I42 I43 I44 I45 I46 I48")

In [3]:
I4 =a
I5 =b
I6 =c
I7 =d
I10 =e
I13 =f
I14 =g
I15 =h
I16 =i
I17 =j
I8 =0.23*I4
I9 =N(I5+0.5*tan(I6*pi/180)*(I4-I8)+0.5*tan(I7*pi/180)*(I4-I8), z)
I11 =N(I10*30/pi*(9.81/(I4/2))**0.5, z)

g =1

I20=N(100*I17/I13/(I17/I13+(100-I17)/1), z)
I21=N(((I4/2)**2*I5*3.14159), z)
I22=N((I20*I13-I20+100*1)/100, z)
I23=N(I15/100*I21*0.6*7.8, z)
I24=N(I13*I21*(I16/100*(0.6+0.4*I20/100)-(I15/100*0.6)), z)
I25=N(I16*I21*0.004*(100-I20)/100*1, z)
I26=N((I24/I13+I23/7.8)/(0.6+0.4*I20/100), z)
I27=N((I23 + I24 + I25)/I26, z)
I28=N(I16/100, z)
I29=N((I20*I13-I20+100)/100, z)
I30=N(I4/2, z)
I32=N(1.57-((1.5708-1.57)*((0.3386+0.1041*I10)+(1.54-2.5673*I10)*I28)), z)



In [4]:
(0.35*(3.364-I28)-(I10))<0

-e - 0.00349807739257812*i + 1.1774 < 0

### Conditions:

* C_1: $ −𝑒−0.0035𝑖+1.18<0$ 
* C_2: $I2<>"grate"$ 
* C_3: $I34>I33$
* C_4: $I33=1.5708$

Steps to follow in the tabular model:
1. check conditions C_1 and C_2
2. calculate I33 and I34
3. check conditions C_3 and C4


Possible combinations:
* C_1, C_2, C_3, C_4 = True, True, True, True
* C_1, C_2, C_3, C_4 = True, False, False, True 
* C_1, C_2, C_3, C_4 = False, False, False, False
* C_1, C_2, C_3, C_4 = False, False, True, False
* C_1, C_2, C_3, C_4 = False, True, False, False
* C_1, C_2, C_3, C_4 = False, True, True, False


In [5]:
C_1, C_2, C_3, C_4 = True, True, True, True



I33 = 1.5708 if C_1 else 2.5307*(1.2796-0.01*𝑖)*(1-exp(-19.42*(0.35*(3.364-0.01*𝑖)-e)))+1.5708
I34 = 3.395 if C_2 else I33
I35 = 0 if not C_2 else (0 if C_3 else (2*3.142*9.81*I5*I29*I11/60)*((I30**3/2*(1-I28))+(I28/3*I30**3)-(I31**2*I30*(1-I28)/2)-(I28*I31**3/3))*(sin(I33)-sin(I34)))
I36 = 0 if C_4 else (I5*I27*(3.142*I11/60*I30/(I30-I38*I31))**3*((I30-I38*I31)**4-((1-I38)*I31)**4))
I39=N((1-(2*3.142*I28/(2*3.142+I32-1.5708)))**0.5,z)
I40=N(((2*3.142+I32-1.5708)/2/3.142*60/((I11)/2))/((I30*(1+I39)*(sin(I32)-sin(1.5708))/9.81)**0.5+((2*3.142+I32-1.5708)/2/3.142*60/((I11)/2))),z)
I31=N(I30*(1-(2*3.142*I40*I28)/((2*3.142+I32-1.5708)))**0.5,z)
I38=N((1-I28)**0.4532,z)
I41=N(4*3.142*I11/60*9.81*(I9-I5)/(I4-I8)*((I27*(sin(I32)-sin(1.5708))+I29*(sin(1.5708)-sin(3.395)))),z)
I42=N(I41*(I30**4-4*I30*I31**3+3*I31**4)/12,z)
I43=N(8*(I9-I5)/(I4-I8)*(I11/60*3.142)**3*(I27),z)
I44=N(I43/20*(I30**5-5*I30*I31**4+4*I31**5),z)
I45=N(I42+I44,z)
I37=N(((3.142*9.81*I5*I27*I11*I30/60)*(2*I30**3-3*I38*I31*I30**2+(3*I38-2)*I31**3)/(3*(I30-I38*I31))*(sin(I32)-sin(1.5708))+0+0+I45),z)
I46=N(1.68*((0.33*I9+0.67*I5)*I4**2.5*I10)**0.82,z)
I48 =N(I37*1.26+I46,z)

In [7]:
print(I48)

14.0*a*b*e*(0.6 + 0.4*j/(f*(-j + 1.0e+2 + j/f)))*(-sin(8.3e-5*e + 8.0e-6*i*(1.5 - 2.6*e) - 1.6) - 1.0)*(-0.38*a**3*(1.0 - 0.01*i)**0.45*(1.0 - 0.003*i*(-0.0008*e - 7.6e-5*i*(1.5 - 2.6*e) + 60.0)*(1/a)**(-0.5)/(e*(0.23*(a*((-0.063*i/(-8.3e-5*e - 8.0e-6*i*(1.5 - 2.6*e) + 6.3) + 1.0)**0.5 + 1.0)*(-sin(8.3e-5*e + 8.0e-6*i*(1.5 - 2.6*e) - 1.6) - 1.0))**0.5 + 0.047*(-0.0008*e - 7.6e-5*i*(1.5 - 2.6*e) + 60.0)*(1/a)**(-0.5)/e)*(-8.3e-5*e - 8.0e-6*i*(1.5 - 2.6*e) + 6.3)))**0.5 + 0.13*a**3*(1.0 - 0.003*i*(-0.0008*e - 7.6e-5*i*(1.5 - 2.6*e) + 60.0)*(1/a)**(-0.5)/(e*(0.23*(a*((-0.063*i/(-8.3e-5*e - 8.0e-6*i*(1.5 - 2.6*e) + 6.3) + 1.0)**0.5 + 1.0)*(-sin(8.3e-5*e + 8.0e-6*i*(1.5 - 2.6*e) - 1.6) - 1.0))**0.5 + 0.047*(-0.0008*e - 7.6e-5*i*(1.5 - 2.6*e) + 60.0)*(1/a)**(-0.5)/e)*(-8.3e-5*e - 8.0e-6*i*(1.5 - 2.6*e) + 6.3)))**1.5*(3.0*(1.0 - 0.01*i)**0.45 - 2.0) + 0.25*a**3)*(0.79*a**2*b*f*(-0.006*h + 0.01*i*(0.6 + 0.4*j/(f*(-j + 1.0e+2 + j/f)))) + 0.037*a**2*b*h + 3.1e-5*a**2*b*i*(1.0e+2 - 1.0e+2*j/(f*(-

$\frac{14.0 a b e \left(0.6 + \frac{0.4 j}{f \left(- j + 1.0 \cdot 10^{2} + \frac{j}{f}\right)}\right) \left(- \sin{\left(8.3 \cdot 10^{-5} e + 8.0 \cdot 10^{-6} i \left(1.5 - 2.6 e\right) - 1.6 \right)} - 1.0\right) \left(- 0.38 a^{3} \left(1.0 - 0.01 i\right)^{0.45} \left(1.0 - \frac{0.003 i \left(- 0.0008 e - 7.6 \cdot 10^{-5} i \left(1.5 - 2.6 e\right) + 60.0\right)}{e \left(0.23 \left(a \left(\left(- \frac{0.063 i}{- 8.3 \cdot 10^{-5} e - 8.0 \cdot 10^{-6} i \left(1.5 - 2.6 e\right) + 6.3} + 1.0\right)^{0.5} + 1.0\right) \left(- \sin{\left(8.3 \cdot 10^{-5} e + 8.0 \cdot 10^{-6} i \left(1.5 - 2.6 e\right) - 1.6 \right)} - 1.0\right)\right)^{0.5} + \frac{0.047 \left(- 0.0008 e - 7.6 \cdot 10^{-5} i \left(1.5 - 2.6 e\right) + 60.0\right)}{e \left(\frac{1}{a}\right)^{0.5}}\right) \left(- 8.3 \cdot 10^{-5} e - 8.0 \cdot 10^{-6} i \left(1.5 - 2.6 e\right) + 6.3\right) \left(\frac{1}{a}\right)^{0.5}}\right)^{0.5} + 0.13 a^{3} \left(1.0 - \frac{0.003 i \left(- 0.0008 e - 7.6 \cdot 10^{-5} i \left(1.5 - 2.6 e\right) + 60.0\right)}{e \left(0.23 \left(a \left(\left(- \frac{0.063 i}{- 8.3 \cdot 10^{-5} e - 8.0 \cdot 10^{-6} i \left(1.5 - 2.6 e\right) + 6.3} + 1.0\right)^{0.5} + 1.0\right) \left(- \sin{\left(8.3 \cdot 10^{-5} e + 8.0 \cdot 10^{-6} i \left(1.5 - 2.6 e\right) - 1.6 \right)} - 1.0\right)\right)^{0.5} + \frac{0.047 \left(- 0.0008 e - 7.6 \cdot 10^{-5} i \left(1.5 - 2.6 e\right) + 60.0\right)}{e \left(\frac{1}{a}\right)^{0.5}}\right) \left(- 8.3 \cdot 10^{-5} e - 8.0 \cdot 10^{-6} i \left(1.5 - 2.6 e\right) + 6.3\right) \left(\frac{1}{a}\right)^{0.5}}\right)^{1.5} \left(3.0 \left(1.0 - 0.01 i\right)^{0.45} - 2.0\right) + 0.25 a^{3}\right) \left(0.79 a^{2} b f \left(- 0.006 h + 0.01 i \left(0.6 + \frac{0.4 j}{f \left(- j + 1.0 \cdot 10^{2} + \frac{j}{f}\right)}\right)\right) + 0.037 a^{2} b h + 3.1 \cdot 10^{-5} a^{2} b i \left(1.0 \cdot 10^{2} - \frac{1.0 \cdot 10^{2} j}{f \left(- j + 1.0 \cdot 10^{2} + \frac{j}{f}\right)}\right)\right) \left(\frac{1}{a}\right)^{0.5}}{\left(- 1.5 a \left(1.0 - 0.01 i\right)^{0.45} \left(1.0 - \frac{0.003 i \left(- 0.0008 e - 7.6 \cdot 10^{-5} i \left(1.5 - 2.6 e\right) + 60.0\right)}{e \left(0.23 \left(a \left(\left(- \frac{0.063 i}{- 8.3 \cdot 10^{-5} e - 8.0 \cdot 10^{-6} i \left(1.5 - 2.6 e\right) + 6.3} + 1.0\right)^{0.5} + 1.0\right) \left(- \sin{\left(8.3 \cdot 10^{-5} e + 8.0 \cdot 10^{-6} i \left(1.5 - 2.6 e\right) - 1.6 \right)} - 1.0\right)\right)^{0.5} + \frac{0.047 \left(- 0.0008 e - 7.6 \cdot 10^{-5} i \left(1.5 - 2.6 e\right) + 60.0\right)}{e \left(\frac{1}{a}\right)^{0.5}}\right) \left(- 8.3 \cdot 10^{-5} e - 8.0 \cdot 10^{-6} i \left(1.5 - 2.6 e\right) + 6.3\right) \left(\frac{1}{a}\right)^{0.5}}\right)^{0.5} + 1.5 a\right) \left(0.0047 a^{2} b h + 0.79 a^{2} b \left(- 0.006 h + 0.01 i \left(0.6 + \frac{0.4 j}{f \left(- j + 1.0 \cdot 10^{2} + \frac{j}{f}\right)}\right)\right)\right)} + 1.7 \left(a^{2.5} e \left(0.13 a \tan{\left(\frac{\pi c}{180} \right)} + 0.13 a \tan{\left(\frac{\pi d}{180} \right)} + 1.0 b\right)\right)^{0.82} + \frac{0.89 e^{3} \left(0.6 + \frac{0.4 j}{f \left(- j + 1.0 \cdot 10^{2} + \frac{j}{f}\right)}\right) \left(3.1 a \tan{\left(\frac{\pi c}{180} \right)} + 3.1 a \tan{\left(\frac{\pi d}{180} \right)}\right) \left(- 0.16 a^{5} \left(1.0 - \frac{0.003 i \left(- 0.0008 e - 7.6 \cdot 10^{-5} i \left(1.5 - 2.6 e\right) + 60.0\right)}{e \left(0.23 \left(a \left(\left(- \frac{0.063 i}{- 8.3 \cdot 10^{-5} e - 8.0 \cdot 10^{-6} i \left(1.5 - 2.6 e\right) + 6.3} + 1.0\right)^{0.5} + 1.0\right) \left(- \sin{\left(8.3 \cdot 10^{-5} e + 8.0 \cdot 10^{-6} i \left(1.5 - 2.6 e\right) - 1.6 \right)} - 1.0\right)\right)^{0.5} + \frac{0.047 \left(- 0.0008 e - 7.6 \cdot 10^{-5} i \left(1.5 - 2.6 e\right) + 60.0\right)}{e \left(\frac{1}{a}\right)^{0.5}}\right) \left(- 8.3 \cdot 10^{-5} e - 8.0 \cdot 10^{-6} i \left(1.5 - 2.6 e\right) + 6.3\right) \left(\frac{1}{a}\right)^{0.5}}\right)^{2.0} + 0.13 a^{5} \left(1.0 - \frac{0.003 i \left(- 0.0008 e - 7.6 \cdot 10^{-5} i \left(1.5 - 2.6 e\right) + 60.0\right)}{e \left(0.23 \left(a \left(\left(- \frac{0.063 i}{- 8.3 \cdot 10^{-5} e - 8.0 \cdot 10^{-6} i \left(1.5 - 2.6 e\right) + 6.3} + 1.0\right)^{0.5} + 1.0\right) \left(- \sin{\left(8.3 \cdot 10^{-5} e + 8.0 \cdot 10^{-6} i \left(1.5 - 2.6 e\right) - 1.6 \right)} - 1.0\right)\right)^{0.5} + \frac{0.047 \left(- 0.0008 e - 7.6 \cdot 10^{-5} i \left(1.5 - 2.6 e\right) + 60.0\right)}{e \left(\frac{1}{a}\right)^{0.5}}\right) \left(- 8.3 \cdot 10^{-5} e - 8.0 \cdot 10^{-6} i \left(1.5 - 2.6 e\right) + 6.3\right) \left(\frac{1}{a}\right)^{0.5}}\right)^{2.5} + 0.031 a^{5}\right) \left(0.79 a^{2} b f \left(- 0.006 h + 0.01 i \left(0.6 + \frac{0.4 j}{f \left(- j + 1.0 \cdot 10^{2} + \frac{j}{f}\right)}\right)\right) + 0.037 a^{2} b h + 3.1 \cdot 10^{-5} a^{2} b i \left(1.0 \cdot 10^{2} - \frac{1.0 \cdot 10^{2} j}{f \left(- j + 1.0 \cdot 10^{2} + \frac{j}{f}\right)}\right)\right) \left(\frac{1}{a}\right)^{1.5}}{a \left(0.0047 a^{2} b h + 0.79 a^{2} b \left(- 0.006 h + 0.01 i \left(0.6 + \frac{0.4 j}{f \left(- j + 1.0 \cdot 10^{2} + \frac{j}{f}\right)}\right)\right)\right)} + \frac{12.0 e \left(0.38 a \tan{\left(\frac{\pi c}{180} \right)} + 0.38 a \tan{\left(\frac{\pi d}{180} \right)}\right) \left(- 0.25 a^{4} \left(1.0 - \frac{0.003 i \left(- 0.0008 e - 7.6 \cdot 10^{-5} i \left(1.5 - 2.6 e\right) + 60.0\right)}{e \left(0.23 \left(a \left(\left(- \frac{0.063 i}{- 8.3 \cdot 10^{-5} e - 8.0 \cdot 10^{-6} i \left(1.5 - 2.6 e\right) + 6.3} + 1.0\right)^{0.5} + 1.0\right) \left(- \sin{\left(8.3 \cdot 10^{-5} e + 8.0 \cdot 10^{-6} i \left(1.5 - 2.6 e\right) - 1.6 \right)} - 1.0\right)\right)^{0.5} + \frac{0.047 \left(- 0.0008 e - 7.6 \cdot 10^{-5} i \left(1.5 - 2.6 e\right) + 60.0\right)}{e \left(\frac{1}{a}\right)^{0.5}}\right) \left(- 8.3 \cdot 10^{-5} e - 8.0 \cdot 10^{-6} i \left(1.5 - 2.6 e\right) + 6.3\right) \left(\frac{1}{a}\right)^{0.5}}\right)^{1.5} + 0.19 a^{4} \left(1.0 - \frac{0.003 i \left(- 0.0008 e - 7.6 \cdot 10^{-5} i \left(1.5 - 2.6 e\right) + 60.0\right)}{e \left(0.23 \left(a \left(\left(- \frac{0.063 i}{- 8.3 \cdot 10^{-5} e - 8.0 \cdot 10^{-6} i \left(1.5 - 2.6 e\right) + 6.3} + 1.0\right)^{0.5} + 1.0\right) \left(- \sin{\left(8.3 \cdot 10^{-5} e + 8.0 \cdot 10^{-6} i \left(1.5 - 2.6 e\right) - 1.6 \right)} - 1.0\right)\right)^{0.5} + \frac{0.047 \left(- 0.0008 e - 7.6 \cdot 10^{-5} i \left(1.5 - 2.6 e\right) + 60.0\right)}{e \left(\frac{1}{a}\right)^{0.5}}\right) \left(- 8.3 \cdot 10^{-5} e - 8.0 \cdot 10^{-6} i \left(1.5 - 2.6 e\right) + 6.3\right) \left(\frac{1}{a}\right)^{0.5}}\right)^{2.0} + 0.063 a^{4}\right) \left(\frac{1.3 j}{- j + 1.0 \cdot 10^{2} + \frac{j}{f}} + \frac{\left(0.6 + \frac{0.4 j}{f \left(- j + 1.0 \cdot 10^{2} + \frac{j}{f}\right)}\right) \left(- \sin{\left(8.3 \cdot 10^{-5} e + 8.0 \cdot 10^{-6} i \left(1.5 - 2.6 e\right) - 1.6 \right)} - 1.0\right) \left(0.79 a^{2} b f \left(- 0.006 h + 0.01 i \left(0.6 + \frac{0.4 j}{f \left(- j + 1.0 \cdot 10^{2} + \frac{j}{f}\right)}\right)\right) + 0.037 a^{2} b h + 3.1 \cdot 10^{-5} a^{2} b i \left(1.0 \cdot 10^{2} - \frac{1.0 \cdot 10^{2} j}{f \left(- j + 1.0 \cdot 10^{2} + \frac{j}{f}\right)}\right)\right)}{0.0047 a^{2} b h + 0.79 a^{2} b \left(- 0.006 h + 0.01 i \left(0.6 + \frac{0.4 j}{f \left(- j + 1.0 \cdot 10^{2} + \frac{j}{f}\right)}\right)\right)} + 1.3 - \frac{1.3 j}{f \left(- j + 1.0 \cdot 10^{2} + \frac{j}{f}\right)}\right) \left(\frac{1}{a}\right)^{0.5}}{a}$